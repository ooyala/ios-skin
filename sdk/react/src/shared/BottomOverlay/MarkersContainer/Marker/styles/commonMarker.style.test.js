// @flow

import {
  calculateLeftPositions, createRootStyle, createTriangleStyle, restrainLeftPositionWithinContainer,
} from './commonMarker.style';

describe('createRootStyle', () => {
  it('returns object matching snapshot', () => {
    expect(createRootStyle()).toMatchSnapshot();
  });
});

describe('createTriangleStyle', () => {
  it('returns object matching snapshot', () => {
    expect(createTriangleStyle()).toMatchSnapshot();
  });
});

describe('restrainLeftPositionWithinContainer', () => {
  it('calculates left offset to place element in the center of container', () => {
    expect(restrainLeftPositionWithinContainer(20, 50, 100)).toEqual(40);
  });

  it('keeps the element within the container on the left side', () => {
    expect(restrainLeftPositionWithinContainer(20, 5, 100)).toEqual(0);
  });

  it('keeps the element within the container on the right side', () => {
    expect(restrainLeftPositionWithinContainer(20, 95, 100)).toEqual(80);
  });
});

describe('calculateLeftPositions', () => {
  it('calculates left offsets for marker and triangle within it', () => {
    expect(calculateLeftPositions(20, 50, 100)).toEqual({
      rootLeft: 40,
      triangleLeft: 4,
    });
  });

  it('balances marker and triangle on the left edge', () => {
    expect(calculateLeftPositions(20, 5, 100)).toEqual({
      rootLeft: -3,
      triangleLeft: 2,
    });
  });

  it('balances marker and triangle on the right edge', () => {
    expect(calculateLeftPositions(20, 95, 100)).toEqual({
      rootLeft: 83,
      triangleLeft: 6,
    });
  });
});
