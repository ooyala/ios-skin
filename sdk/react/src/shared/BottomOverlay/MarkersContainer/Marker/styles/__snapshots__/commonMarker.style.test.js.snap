// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createRootStyle returns object matching snapshot 1`] = `
Object {
  "backgroundColor": "black",
  "borderColor": "#4E4E4E",
  "borderRadius": 2,
  "borderWidth": 1,
  "bottom": 6,
  "padding": 2,
  "position": "absolute",
  "shadowColor": "black",
  "shadowOffset": Object {
    "height": 2,
    "width": 0,
  },
  "shadowOpacity": 0.5,
  "shadowRadius": 4,
}
`;

exports[`createTriangleStyle returns object matching snapshot 1`] = `
Object {
  "borderBottomWidth": 0,
  "borderColor": "transparent",
  "borderTopColor": "#4E4E4E",
  "borderWidth": 6,
  "bottom": -6,
  "height": 0,
  "position": "absolute",
  "width": 0,
}
`;
